source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/TentacleBinaryCache.cs

```diff
@@ -24,47 +24,63 @@ public TentacleBinaryCache(ITentacleFetcher tentacleFetcher, ILogger logger)
 
         private static readonly string cacheDirRunExtension = Guid.NewGuid().ToString("N");
 
-        private static readonly ConcurrentDictionary<Version, SemaphoreSlim> versionLock = new();
+        private static readonly ConcurrentDictionary<(Version, TentacleRuntime), SemaphoreSlim> versionLock = new();
 
-        public async Task<string> GetTentacleVersion(string tmp, Version version, CancellationToken cancellationToken)
+        // TODO: Fun times with Runtimes
+        public async Task<string> GetTentacleVersion(string tmp, Version version, TentacleRuntime runtime, CancellationToken cancellationToken)
         {
-            using var _ = await versionLock.GetOrAdd(version, s => new SemaphoreSlim(1, 1)).LockAsync();
+            using var _ = await versionLock.GetOrAdd((version, runtime), s => new SemaphoreSlim(1, 1)).LockAsync();
+            
+            var runtimeString = runtime.GetStringValue();
```

Comment:
nit: could you achieve this by overriding .ToStrig()

Reply:
Or should it be "GetDescription" as I see you also rely on .ToString below so wouldn't want to  change the default behavour

Reply:
We can't override ToString on an `enum`. We could create a type-safe enum for the runtime, then override `ToString`, but I don't think it's worth the rework.

I did rename the extension method to `GetDescription` though, as that makes the intent clearer. I also got rid of my custom attribute, and just use `System.ComponentModel.DescriptionAttribute` instead.